src/main/java/sirius/db/jdbc/SmartQuery.java

```diff
@@ -366,25 +366,54 @@ private List<E> queryNextBlock() {
             if (lastValue == null) {
                 return effectiveQuery.queryList();
             }
-            CompoundValue leftHandSide = new CompoundValue();
-            CompoundValue rightHandSide = new CompoundValue();
-
+            SQLConstraint sortingFilterConstraint = null;
+            Map<Mapping, Object> previousSortingColumns = new HashMap<>();
             for (Tuple<Mapping, Boolean> sorting : effectiveQuery.orderBys) {
                 Mapping sortColumn = sorting.getFirst();
                 boolean sortAscending = sorting.getSecond().booleanValue();
                 Object value = getPropertyValue(sortColumn, lastValue);
-                if (sortAscending) {
-                    // the order by is ascending -> COLUMN > lastvalue.column
-                    leftHandSide.addComponent(sortColumn);
-                    rightHandSide.addComponent(value);
-                } else {
-                    // the order by is descending -> lastvalue.column > COLUMN
-                    leftHandSide.addComponent(value);
-                    rightHandSide.addComponent(sortColumn);
-                }
+
+                SQLConstraint currentColumConstraint = createSqlConstraintForSortingColumn(sortAscending, sortColumn,
+                        value, previousSortingColumns);
+                sortingFilterConstraint = OMA.FILTERS.or(sortingFilterConstraint, currentColumConstraint);
+
+                previousSortingColumns.put(sortColumn, value);
             }
+            return effectiveQuery.where(sortingFilterConstraint).queryList();
+        }
 
-            return effectiveQuery.where(OMA.FILTERS.gt(leftHandSide, rightHandSide)).queryList();
+        /**
+         * Creates a sql constraint for sorting purposes.
+         * </p>
+         * In MySQL/MariaDB, NULL is considered as a 'missing, unkonwn value'. Any arithmetic comparison with NULL
```

Comment:
is this mysql specific or according to SQL standard? if this would be specific we should tie this to a database capability...

Reply:
This is basically SQL Standard, but there are also implementation specific differences...

Reply:
More precisely: `NULL != 'any'` => false is SQL standard. However, the sort order with null values (nulls first vs last) is implementation specific